extra/provision.sh

```diff
@@ -1,9 +1,34 @@
 #!/bin/bash
 #
-# Facebook CTF: Provision script for vagrant dev environment
+# fbctf provisioning script
 #
-# Usage: ./provision.sh [dev | prod] [path_to_code] [path_to_destination]
+# Usage: usage [-h|--help] [PARAMETER [ARGUMENT]] [PARAMETER [ARGUMENT]] ...
 #
+# Parameters:
+#   -h, --help            Shows this help message and exit.
+#   -m MODE, --mode MODE  Mode of operation. Default value is dev
+#   -c TYPE, --cert TYPE  Type of certificate to use. Default value is self
+#
+# Arguments:
+#   dev    Provision will run in development mode. Certificate will be self-signed.
+#   prod   Provision will run in production mode.
+#   self   Provision will use a self-signed SSL certificate that will be generated.
+#   own    Provision will use the SSL certificate provided by the user.
+#   certbot Provision will generate a SSL certificate using letsencrypt/certbot. More info here: https://certbot.eff.org/
+#
+# Optional Parameters:
+#   -k PATH, --keyfile PATH      Path to supplied SSL key file.
+#   -C PATH, --certfile PATH     Path to supplied SSL certificate pem file.
+#   -d DOMAIN, --domain DOMAIN   Domain for the SSL certificate to be generated using letsencrypt.
```

Comment:
Let's make this capital `D` since we have capital `C` here as well. For source and dest we can use lower case letters (`s` and `d`).